File: Main2_14.java
import java.util.Scanner;

public class Main2_14 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int result = 0;
        int max = Integer.MIN_VALUE;
        int min = sc.nextInt();

        for (int i = 0; i < n - 1; i++) {
            int x = sc.nextInt();

            result = x - min;

            if (result > max) {
                max = result;
            }

            if (x < min) {
                min = x;
            }
        }

        System.out.println(max);
        sc.close();
    }
}

File: Main2_14.py
def main():
    n = int(input())

    max_value = -10**18
    min_value = int(input())

    for _ in range(n - 1):
        x = int(input())

        result = x - min_value

        if result > max_value:
            max_value = result

        if x < min_value:
            min_value = x

    print(max_value)


if __name__ == "__main__":
    main()